### PR TITLE
Multiplication and subtraction methods

### PR DESCRIPTION
**Pull request by julianmateu**
On this PR I added methods and their respective tests for:
1. Subtract one quaternion from another.
2. Multiply one quaternion by a scalar.
3. Multiply two quaternions and find their Hamilton product.

@basicNew

### CODE REVIEW

File: src/Quaternion.java [hunk 1]
```diff
@@ -20,75 +24,151 @@ public Quaternion(double r, double i, double j, double k) {
...
+
+    /**
+     * Method to add two Quaternions.
+     *
+     * @return It returns a new Quaternion with the result of the sum.
      */
-    public double getR() {return this.r;}
+    public Quaternion add(Quaternion addend) {
+        return new Quaternion(
+                this.r + addend.getR(),
```
Comment (basicNew) [inline src/Quaternion.java]: use getters to encapsulate yourself form the implementation

File: src/Quaternion.java [hunk 2]
```diff
@@ -20,75 +24,151 @@ public Quaternion(double r, double i, double j, double k) {
...
-    /** i Getter
+    /**
+     * Method to subtract two Quaternions.
+     *
+     * @return It returns a new Quaternion with the result of the subtraction.
      */
-    public double getI() {return this.i;}
+    public Quaternion subtract(Quaternion subtrahend) {
+        return new Quaternion(
+                this.r - subtrahend.getR(),
```
Comment (basicNew) [inline src/Quaternion.java]: same as above

File: src/Quaternion.java [hunk 3]
```diff
@@ -20,75 +24,151 @@ public Quaternion(double r, double i, double j, double k) {
...
-    /** k Getter
+    /**
+     * Method to assert if to Quaternions are equal.
+     *
+     * @return Returns True if they are equal, and False otherwise.
      */
-    public double getK() {return this.k;}
+    public boolean equals(Quaternion compared) {
+
+        return this.r == compared.getR() && this.i == compared.getI() && this.j ==
```
Comment (basicNew) [inline src/Quaternion.java]: same as above

File: src/Quaternion.java [hunk 4]
```diff
@@ -20,75 +24,151 @@ public Quaternion(double r, double i, double j, double k) {
...
-    /** r Setter
+    /**
+     * Method to convert a Quaternion to a string.
+     *
+     * @return Returns the Quaternion in the form ( r ; i ; j ; k ).
      */
-    public void setR(double r) {this.r = r;}
+    @Override
+    public String toString() {
+        return "( " + this.r + " ; " + this.i + " ; " + this.j + " ; " + this.k + " )";
```
Comment (basicNew) [inline src/Quaternion.java]: same as above

File: src/Quaternion.java [hunk 5]
```diff
@@ -20,75 +24,151 @@ public Quaternion(double r, double i, double j, double k) {
...
 
-    /** Method to convert a Quaternion to a string.
-     * @return Returns the Quaternion in the form ( r ; i ; j ; k ).
+    /**
+     * k Setter.
      */
-    @Override
-    public String toString(){
-        return "( " + this.r + " ; " + this.i + " ; " + this.j + " ; " + this.k + " )";
+    public void setK(double k) {
```
Comment (basicNew) [inline src/Quaternion.java]: Do we need public setters?

File: src/TestQuaternion.java [hunk 6]
```diff
@@ -1,49 +1,183 @@
...
+        // Simple tests:
+        assertEquals(new Quaternion(1, 2, 3, 4).multiply(new Quaternion(2, 2, 2, 2)),
+                new Quaternion(-16, 4, 12, 8));
+        assertEquals(new Quaternion(1, 2, 3, 4).multiply(new Quaternion(1, 2, 3, 4)),
+                new Quaternion(-28, 4, 6, 8));
+        assertEquals(new Quaternion(1, 1, 1, 1).multiply(new Quaternion(0, 0, 0, 0)),
+                new Quaternion(0, 0, 0, 0)); // Multiplying by right zero element.
+        assertEquals(new Quaternion(0, 0, 0, 0).multiply(new Quaternion(1, 1, 1, 1)),
+                new Quaternion(0, 0, 0, 0)); // Multiplying by left zero element.
+        assertEqualsQuaternion(new Quaternion(1, 2, 3, 4).multiply(new Quaternion(
```
Comment (basicNew) [inline src/TestQuaternion.java]: you may want to use tempoaral variables in these cases, as the test is a bit hard to read

**Comment by basicNew** [comment]
Just a few comments, otherwise looks good @julianmateu ! Let me know when those are addressed and I'll merge

**Comment by julianmateu** [comment]
@basicNew I think I've corrected the mistakes now.

**Pull request merged**